omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/TaskActionController1_8Test.java

```diff
@@ -0,0 +1,145 @@
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.TaskActionResource1_8;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.TaskDefinitionResource1_8;
+import org.openmrs.scheduler.TaskDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import java.util.Arrays;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsNot.not;
+
+public class TaskActionController1_8Test extends MainResourceControllerTest {
+	
+	@Autowired
+	RestService restService;
+	
+	private TaskDefinition testTask = new TaskDefinition(1, "TestTask", "TestTask Description",
+	        "org.openmrs.scheduler.tasks.TestTask");
+	
+	private TaskDefinition tempTask = new TaskDefinition(3, "TempTask", "TempTask Description",
+	        "org.openmrs.scheduler.tasks.TestTask");
+	
+	private MockTaskServiceWrapper mockTaskServiceWrapper = new MockTaskServiceWrapper();
+	
+	@Before
+	public void setUp() throws Exception {
+		mockTaskServiceWrapper.registeredTasks.addAll(Arrays.asList(testTask, tempTask));
+		
+		TaskActionResource1_8 taskActionResource = (TaskActionResource1_8) restService
+		        .getResourceBySupportedClass(TaskAction.class);
+		taskActionResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+		
+		TaskDefinitionResource1_8 taskResource = (TaskDefinitionResource1_8) restService
+		        .getResourceBySupportedClass(TaskDefinition.class);
+		taskResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+	}
+	
+	@Test
+	public void shouldScheduleTask() throws Exception {
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask)));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"scheduletask\", \"tasks\":[\"" + getTestTaskName()
+		        + "\"]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask));
+	}
+	
+	@Test
+	public void shouldShutdownTask() throws Exception {
+		mockTaskServiceWrapper.scheduledTasks.add(testTask);
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"shutdowntask\", \"tasks\":[\"" + getTestTaskName()
+		        + "\"]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask)));
+	}
+	
+	@Test
+	public void shouldDoNothingIfTaskAlreadyScheduled() throws Exception {
+		mockTaskServiceWrapper.scheduledTasks.add(testTask);
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"scheduletask\", \"tasks\":[\"" + getTestTaskName()
+		        + "\"]}")));
+		//check if state preserved
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask));
+	}
+	
+	@Test
+	public void shouldDoNothingIfTaskAlreadyShutdown() throws Exception {
```

Comment:
This should be: shutdown_shouldDoNothingIfTaskAlreadyShutdown

Reply:
Changed